src/ctx.rs

```diff
@@ -23,54 +78,95 @@ struct PersistedVariable {
     default_value: serde_yaml::Value,
 }
 
-impl TryFrom<PersistedOptions> for Variables {
+impl TryFrom<Vec<PersistedVariable>> for Variables {
     type Error = crate::Error;
-    fn try_from(persisted: PersistedOptions) -> Result<Self> {
+    fn try_from(persisted: Vec<PersistedVariable>) -> Result<Self> {
         let mut out = Variables::default();
-        for saved_var in persisted.variables {
+        for saved_var in persisted {
             out.insert(saved_var.name, saved_var.default_value)?;
         }
         Ok(out)
     }
 }
 
-impl From<Variables> for PersistedOptions {
+impl From<Variables> for Vec<PersistedVariable> {
     fn from(variables: Variables) -> Self {
-        let formatted_variables = variables
+        variables
             .tree()
             .iter()
             .map(|(k, v)| PersistedVariable {
                 name: k.into(),
                 default_value: v.clone(),
             })
-            .collect::<Vec<PersistedVariable>>();
-        PersistedOptions {
-            variables: formatted_variables,
-        }
+            .collect::<Vec<PersistedVariable>>()
     }
 }
 
-pub fn extract_variables(ctx: &Ctx) -> Result<(Variables, Variables)> {
-    let mut confirmed_variables = Variables::default();
-    confirmed_variables.insert(
+pub(crate) struct VariablesFromCtx {
+    pub src: Variables,
+    pub cli: Variables,
+    pub saved: Variables,
+}
+
+pub(crate) fn extract_variables(ctx: &Ctx) -> Result<VariablesFromCtx> {
+    let mut default_variables = Variables::default();
+    default_variables.insert(
         "ffizer_dst_folder",
         ctx.cmd_opt
             .dst_folder
             .to_str()
             .expect("dst_folder to converted via to_str"),
     )?;
-    confirmed_variables.insert("ffizer_src_uri", ctx.cmd_opt.src.uri.raw.clone())?;
-    confirmed_variables.insert("ffizer_src_rev", ctx.cmd_opt.src.rev.clone())?;
-    confirmed_variables.insert("ffizer_src_subfolder", ctx.cmd_opt.src.subfolder.clone())?;
-    confirmed_variables.insert("ffizer_version", env!("CARGO_PKG_VERSION"))?;
+    default_variables.insert("ffizer_src_uri", ctx.cmd_opt.src.uri.raw.clone())?;
+    default_variables.insert("ffizer_src_rev", ctx.cmd_opt.src.rev.clone())?;
+    default_variables.insert("ffizer_src_subfolder", ctx.cmd_opt.src.subfolder.clone())?;
+    default_variables.insert("ffizer_version", env!("CARGO_PKG_VERSION"))?;
 
-    confirmed_variables.append(&mut get_cli_variables(ctx)?);
+    let confirmed_variables = get_cli_variables(ctx)?;
     let suggested_variables = get_saved_variables(ctx)?;
 
-    Ok((confirmed_variables, suggested_variables))
+    Ok(VariablesFromCtx {
+        src: default_variables,
+        cli: confirmed_variables,
+        saved: suggested_variables,
+    })
```

Comment:
tips: in rust if the variable has the same name has the field you can replace 

```rust
Toto {
  foo: foo,
}
```

 by 
 
```rust
Toto {
  foo,
}
```